### PR TITLE
3 feature dockerize the project

### PR DESCRIPTION
**Pull request by bigbeef1994**
# Summary

### Related Issues

- Completes #3 

### Contributors

- @alexander-ma - implementer and tester
- @bigbeef1994 - implementer

### Affected Modules

- Dockerfile
- docker-compose.yml 

# Description

Alex and Daniel worked together to get Docker working with this project. 

Alex first created a Dockefile following examples from previous projects. The Dockerfile specifies the base image as golang:latest and also sets the working directory to the project directory. Next, the Dockerfile populates the caches based on GO dependencies. The Dockerfile also specifies unit tests and then builds the executable for the image. The final command from the Dockerfile is to expose port 8080. 

Daniel created the docker-compose.yml file. The purpose of the compose file is to prop up the Dockerfile previously created. The compose file specifies the webapp to be hosted on port 8080 which was exposed in the Dockerfile. The compose file also sets up the relational database that will be used for this project, which in this case is postgres. The database instance is run on port 5432 and has simple username and password for access. 

# Testing (Unit/Integration/Performance)

Alex performed simple tests, both on the Dockerfile and the compose file. 

In validating the Dockerfile, Alex built an image and then ran that image on a container. Using APIs defined by Danny, Alex checked the health of the server with the health endpoint. 

Alex validated the compose file and found a bug in defining the port, which he fixed immediately. He was able to both obtain a healthy response from the webapp as he had previously done with the Dockerfile and start a postgres instance using pgcli.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,4 +16,4 @@ coverage.txt
 *.out
 
 # Dependency directories (remove the comment below to include it)
-# vendor/
+# vendor/
```
Comment (dvo-dev) [inline .gitignore:19]: missing EOF (add blank line)

File: .gitignore [hunk 2]
```diff
@@ -6,6 +6,7 @@
 *.dylib
 bin/
 .vscode/
+.DS_Store
```
Comment (dvo-dev) [inline .gitignore:9]: pog

File: Makefile [hunk 3]
```diff
@@ -25,4 +25,8 @@ run: build
 	$(OBJDIR)/$(BINARY)
 
 clean:
-	rm -rf $(OUTDIR)
\ No newline at end of file
+	rm -rf $(OUTDIR)
+
+make db-pgcli:
```
Comment (dvo-dev) [inline Makefile:30]: bro take out that `make `

Comment (bigbeef1994) [reply Makefile:30]: pog

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+version: "3.9"
+services:
+  web:
+    image: go-get-started:latest
```
Comment (dvo-dev) [inline docker-compose.yml:4]: did you try running this yourself? this is attempting to fetch from docker repository instead of using the local Dockerfile

**Review by dvo-dev — APPROVED** [review]
LGTM

**Pull request merged**